modules/channel-messenger/src/backend/messenger.ts

```diff
@@ -0,0 +1,253 @@
+import axios, { AxiosInstance } from 'axios'
+import bodyParser from 'body-parser'
+import * as sdk from 'botpress/sdk'
+import crypto from 'crypto'
+import { Router } from 'express'
+import _ from 'lodash'
+
+import { Config } from '../config'
+
+const outgoingTypes = ['text', 'typing', 'login_prompt', 'carousel']
+type MessengerAction = 'typing_on' | 'typing_off' | 'mark_seen'
+
+export class MessengerService {
+  private readonly http = axios.create({ baseURL: 'https://graph.facebook.com/v2.6/me' })
+
+  private messengerClients: { [key: string]: MessengerClient } = {}
+  private router: Router
+
+  constructor(private bp: typeof sdk) {}
+
+  initialize() {
+    this.router = this.bp.http.createRouterForBot('channel-messenger', { checkAuthentication: false })
+    this.router.use(
+      bodyParser.json({
+        verify: this._verifySignature.bind(this)
+      })
+    )
+
+    this.router.get('/webhook', this._setupWebhook.bind(this))
+    this.router.post('/webhook', this._handleIncomingMessage.bind(this))
+
+    this.bp.events.registerMiddleware({
+      description: 'Sends outgoing messages for the messenger channel',
+      direction: 'outgoing',
+      handler: this._handleOutgoingEvent.bind(this),
+      name: 'messenger.sendMessages',
+      order: 200
+    })
+  }
+
+  getMessengerClient(botId: string): MessengerClient {
+    if (this.messengerClients[botId]) {
+      return this.messengerClients[botId]
+    }
+
+    this.messengerClients[botId] = new MessengerClient(botId, this.bp, this.http)
```

Comment:
This creates a sperfluous client even if it's not enabled for the bot. Please use the bot mount/unmount logic to create the clients

Reply:
The client is created when Facebook verify the webhook so it is not actually created for all bots.